editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -129,41 +197,39 @@ export function findReparentStrategies(
   const metadata = canvasState.startingMetadata
 
   const reparentSubjects =
-    canvasState.interactionTarget.type === 'INSERTION_SUBJECTS' &&
-    canvasState.interactionTarget.subjects.length > 0
+    canvasState.interactionTarget.type === 'INSERTION_SUBJECTS'
       ? newReparentSubjects(canvasState.interactionTarget.subjects[0].defaultSize)
       : existingReparentSubjects(
           getDragTargets(getTargetPathsFromInteractionTarget(canvasState.interactionTarget)), // uhh
         )
 
-  const getReparentTargetInner = (missingBoundsHandling: MissingBoundsHandling) =>
-    getReparentTargetUnified(
-      reparentSubjects,
-      pointOnCanvas,
-      cmdPressed,
-      canvasState,
-      metadata,
-      canvasState.startingAllElementProps,
-      missingBoundsHandling,
-    )
+  const targetParent = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    cmdPressed,
+    canvasState,
+    metadata,
+    canvasState.startingAllElementProps,
+    'allow-missing-bounds', // TODO delete this property!
+  )
 
-  const strictTarget = getReparentTargetInner('use-strict-bounds')
-  const strictStrategy =
-    strictTarget == null ? null : reparentStrategyForReparentTarget(metadata, strictTarget, false)
-
-  const forcedTarget = getReparentTargetInner('allow-missing-bounds')
-  const sameTargets =
-    strictTarget != null &&
-    forcedTarget != null &&
-    EP.pathsEqual(forcedTarget.newParent, strictTarget.newParent)
-  const convertToAbsolute = sameTargets && strictStrategy?.strategy === 'REPARENT_TO_FLEX'
-  const skipForcedTarget = sameTargets && !convertToAbsolute
-  const forcedStrategy =
-    forcedTarget == null || skipForcedTarget
-      ? null
-      : reparentStrategyForReparentTarget(metadata, forcedTarget, convertToAbsolute)
+  if (targetParent == null) {
+    return [] // TODO is canvas a null target parent?
+  }
+
+  const strategy = reparentStrategyForReparentTarget(metadata, targetParent, false)
+
+  const fallbackStrategy: FindReparentStrategyResult =
+    strategy.strategy === 'REPARENT_AS_ABSOLUTE'
+      ? {
+          isFallback: true, // TODO is this true?
```

Comment:
I'm very confused about what's happening here and why!

Reply:
I added some clarification comments in the code, and removed the TODO questions

Reply:
Thank you!